src/main/java/io/github/sashirestela/openai/BaseSimpleOpenAI.java

```diff
@@ -0,0 +1,119 @@
+package io.github.sashirestela.openai;
+
+import io.github.sashirestela.cleverclient.CleverClient;
+import java.net.http.HttpClient;
+import java.util.Optional;
+import lombok.NonNull;
+import lombok.Setter;
+
+
+/**
+ * The base abstract class that all providers extend
+ */
+
+
+public class BaseSimpleOpenAI {
```

Comment:
I don't think so. It can be instantiated on its own. For example, if someone wants to access another OpenAI provider that is not one of the officially supported: OpenAI, Azure OpenAI or Anyscale. They can directly proper create BaseSimpleOpenAIArgs and instantiate a BaseSimpleOpenAI object.

Reply:
Please, turn into an abstract class. Any other provider must extend this abstract class